### PR TITLE
rgw: Refactoring Quota Management

### PR DESCRIPTION
**Pull request by iqbalredkhan**
## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/rgw/rgw_quota.cc [hunk 1]
```diff
@@ -904,12 +904,12 @@ class RGWQuotaHandlerImpl : public RGWQuotaHandler {
   int check_quota(const DoutPrefixProvider *dpp,
                   const rgw_user& user,
 		  rgw_bucket& bucket,
-		  RGWQuotaInfo& user_quota,
-		  RGWQuotaInfo& bucket_quota,
+      RGWQuota& quota,
```
Comment (alimaredia) [inline src/rgw/rgw_quota.cc:907]: Nit: fix the indentation here

Comment (iqbalredkhan) [reply src/rgw/rgw_quota.cc]: done

**Comment by alimaredia** [comment]
jenkins test make check

**Comment by alimaredia** [comment]
jenkins test api

File: src/rgw/rgw_op.cc [hunk 2]
```diff
@@ -1357,18 +1357,20 @@ int RGWOp::init_quota()
...
 
-  store->get_quota(bucket_quota, user_quota);
+  store->get_quota(quota);
 
   if (s->bucket->get_info().quota.enabled) {
-    bucket_quota = s->bucket->get_info().quota;
-  } else if (user->get_info().bucket_quota.enabled) {
-    bucket_quota = user->get_info().bucket_quota;
+    quota.bucket_quota = s->bucket->get_info().quota;
+
```
Comment (dang) [inline src/rgw/rgw_op.cc:1367]: Extra blank line

File: src/rgw/rgw_op.cc [hunk 3]
```diff
@@ -5471,8 +5473,7 @@ void RGWCopyObj::execute(optional_yield y)
         return;
       }
       // enforce quota against the destination bucket owner
-      op_ret = dest_bucket->check_quota(this, user_quota, bucket_quota,
-				      astate->accounted_size, y);
+      op_ret = dest_bucket->check_quota(this, quota,astate->accounted_size, y);
```
Comment (dang) [inline src/rgw/rgw_op.cc:5476]: missing space after quota,

**Review by dang — COMMENTED** [review]
Looks great.  A few minor whitespace nits.

**Comment by alimaredia** [comment]
@iqbalredkhan this is minor but could you add a commit message describing what we you did in this PR. Also in the title you misspelled `Refactoring` `refactoing`.

**Comment by iqbalredkhan** [comment]
jenkins test api

File: src/rgw/rgw_op.cc (hunk 2 above)
Comment (iqbalredkhan) [reply src/rgw/rgw_op.cc]: done

File: src/rgw/rgw_op.cc (hunk 3 above)
Comment (iqbalredkhan) [reply src/rgw/rgw_op.cc]: did

**Comment by iqbalredkhan** [comment]
> @iqbalredkhan this is minor but could you add a commit message describing what we you did in this PR. Also in the title you misspelled `Refactoring` `refactoing`.

I updated the commit as well as heading

**Comment by iqbalredkhan** [comment]
Here is the result of teuthology run
https://pulpito.ceph.com/iqbalredkhan-2022-04-29_08:52:40-rgw-wip-iqbal-refactoring-quota-management-distro-default-smithi/

**Review by alimaredia — APPROVED** [review]

**Comment by alimaredia** [comment]
jenkins test api

**Pull request merged**